packages/nimble-blazor/Tests/NimbleBlazor.Tests.Acceptance/Tests/TableTests.cs

```diff
@@ -24,5 +24,45 @@ await using (var pageWrapper = await NewPageForRouteAsync("TableSetData"))
                 await Assertions.Expect(rows).ToContainTextAsync(new string[] { "A0", "A1", "A2", "A3", "A4" });
             }
         }
+
+        [Theory]
+        [InlineData(0, true, false)]
+        [InlineData(1, false, false)]
+        [InlineData(2, false, true)]
+        [InlineData(3, false, false)]
```

Comment:
I presume the test is structured this way so we don't do assertions within a loop? Can't think of a real alternative to this (this initially felt odd to me as we are making a theory-based test that doesn't actually alter any state based on the theory data).

Reply:
Yes, you are correct. I created it like this to have a test per row, which corresponds to a test per possible delayed hierarchy state (`canLoadChildren`, `loadingChildren`, `none`, and not set), rather than a single test that loops through checking each row.